homeassistant/components/rainmachine/switch.py

```diff
@@ -164,6 +198,28 @@ async def async_setup_entry(
             },
             "async_start_zone",
         ),
+        (
+            "push_weather_data",
+            {
+                vol.Optional(ATTR_TIMESTAMP, default=None): vol.Any(
+                    cv.positive_float, None
+                ),
+                vol.Optional(ATTR_MINTEMP, default=None): CV_VALID_TEMP_RANGE,
+                vol.Optional(ATTR_MAXTEMP, default=None): CV_VALID_TEMP_RANGE,
+                vol.Optional(ATTR_TEMPERATURE, default=None): CV_VALID_TEMP_RANGE,
+                vol.Optional(ATTR_WIND, default=None): CV_VALID_WIND_SPEED,
+                vol.Optional(ATTR_SOLARRAD, default=None): CV_VALID_SOLARRAD,
+                vol.Optional(ATTR_QPF, default=None): CV_VALID_RAIN_RANGE,
+                vol.Optional(ATTR_RAIN, default=None): CV_VALID_RAIN_RANGE,
+                vol.Optional(ATTR_ET, default=None): CV_VALID_RAIN_RANGE,
+                vol.Optional(ATTR_MINRH, default=None): CV_VALID_PERCENTAGE,
+                vol.Optional(ATTR_MAXRH, default=None): CV_VALID_PERCENTAGE,
+                vol.Optional(ATTR_CONDITION, default=None): vol.Any(str, None),
```

Comment:
Do we need `vol.Any(str, None)`? I would think that since we mark this field as optional, `None` is already a valid value. Wondering if this would work:

```python
vol.Optional(ATTR_CONDITION, default=None): cv.string
```

Reply:
Good point. I will test tomorrow, but I think it just needs to be:

```python 
vol.Optional(ATTR_CONDITION): cv.string
```

It's been a minute since I've messed with voluptuous.